### ISSUE TITLE
404 Not Found with Get frames request.

### ISSUE THREAD

**Issue by taivnn9** [dcmjs-org/dicomweb-server#88]
**I have using POST dcm4che stowrs to upload my dicom file, process was success.**
![image](https://user-images.githubusercontent.com/24297201/225958444-25530c65-0301-4cbe-a3d1-85fb664f48d4.png)
![image](https://user-images.githubusercontent.com/24297201/225958837-d572fbc9-cc30-4c41-bedd-67d2c5590105.png)

**Study already on my server & DB.**
![image](https://user-images.githubusercontent.com/24297201/225958988-a8fdf4a9-0d04-4cd7-9687-e072e1f1bebf.png)
![image](https://user-images.githubusercontent.com/24297201/225959104-185f1548-8fd9-4d6f-bd9d-cf9dc53f2b13.png)

**But get frames not work.**
![image](https://user-images.githubusercontent.com/24297201/225959492-e7f17921-31ce-4ffc-987d-908498c886ff.png)
![image](https://user-images.githubusercontent.com/24297201/225959691-627c72f4-6deb-45fc-9d27-2483a3f8da64.png)

Please tell me why^^ thank for help

**Comment by pieper** [comment]
You can check with public data to see if there's something unique about the data you are using.  You can also try accessing the same frame endpoint using curl to see if there's a server side issue.

**Comment by taivnn9** [comment]
Hi, my dicom file download from `https://www.rubomedical.com/dicom_files/dicom_viewer_0003.zip`, it works well with Radiant & Weasis,
I also try with Postman and have same status response code.
Here is DicomCloud log
```
Application Insights Telemetry (unconfigured): {"name":"Microsoft.ApplicationInsights.Dev.Request","time":"2023-03-17T16:59:22.9846635Z","tags":{"ai.cloud.roleInstance":"DESKTOP-ODVQOLT","ai.operation.id":"312f2279553ef04e9c8de597aae511a2","ai.operation.name":"GET /api/studies/1.3.12.2.1107.5.4.3.123456789012345.19950922.121803.6/series/1.3.12.2.1107.5.4.3.123456789012345.19950922.121803.8/instances/1.3.12.2.1107.5.4.3.321890.19960124.162922.29/frames/1","ai.location.ip":"::1","ai.internal.sdkVersion":"web:2.8.1-19196","ai.internal.nodeName":"DESKTOP-ODVQOLT"},"data":{"baseType":"RequestData","baseData":{"ver":2,"id":"|312f2279553ef04e9c8de597aae511a2.3c6e5c00_","name":"GET /api/studies/1.3.12.2.1107.5.4.3.123456789012345.19950922.121803.6/series/1.3.12.2.1107.5.4.3.123456789012345.19950922.121803.8/instances/1.3.12.2.1107.5.4.3.321890.19960124.162922.29/frames/1","duration":"00:00:00.1889790","success":false,"responseCode":"406","url":"http://localhost:44301/api/studies/1.3.12.2.1107.5.4.3.123456789012345.19950922.121803.6/series/1.3.12.2.1107.5.4.3.123456789012345.19950922.121803.8/instances/1.3.12.2.1107.5.4.3.321890.19960124.162922.29/frames/1","properties":{"DeveloperMode":"true","_MS.ProcessedByMetricExtractors":"(Name:'Requests', Ver:'1.1')"}}}}
The thread 0x36e0 has exited with code 0 (0x0).
Application Insights Telemetry (unconfigured): {"name":"Microsoft.ApplicationInsights.Dev.Metric","time":"2023-03-17T16:59:01.0000000Z","tags":{"ai.cloud.roleInstance":"DESKTOP-ODVQOLT","ai.internal.sdkVersion":"m-agg2:2.8.1-22898"},"data":{"baseType":"MetricData","baseData":{"ver":2,"metrics":[{"name":"Server response time","kind":"Aggregation","value":188.979,"count":1,"min":188.979,"max":188.979,"stdDev":0}],"properties":{"DeveloperMode":"true","_MS.MetricId":"requests/duration","_MS.IsAutocollected":"True","_MS.AggregationIntervalMs":"60000","Request.Success":"False"}}}}
``` 
Here is Postman request header
![image](https://user-images.githubusercontent.com/24297201/225970678-303e7761-be9f-4de0-9b70-f7d66c3bc731.png)

Here is DicomCloud debugger.
![image](https://user-images.githubusercontent.com/24297201/225971370-3e9dc893-ffe1-40e2-af1d-3e8683c447c5.png)

**Comment by taivnn9** [comment]
Hi. My Dicom file work fine with Weasis & RadiAnt `https://www.rubomedical.com/dicom_files/dicom_viewer_0003.zip`

I debug that exception in here, do you know why? thank!
> ![image](https://user-images.githubusercontent.com/24297201/225989865-ba338952-75c5-4f88-8e20-6855e33c5296.png)
> 
![image](https://user-images.githubusercontent.com/24297201/225990036-296040f5-2376-4677-a325-3ba5de7e60ea.png)

**Comment by pieper** [comment]
hard to tell from here, but it looks like you are on the right path debugging.  See if you can trace through data that works and compare what happens with this data.

**Comment by taivnn9** [comment]
I think that API trying to download dcm file via http request and they not found the resource,
what does this mean "trace through data that works"?

**Comment by pieper** [comment]
> what does this mean "trace through data that works"?

I mean that the server should be working generally for sample data, so you can use a debugger to step through a case where it works vs a case where it doesn't work and compare the difference.  If it's not working for any data then perhaps there's something wrong with the installation.

**Comment by taivnn9** [comment]
Unfortunately, this happens with all my datasets. :(  I reproduce with many dicom files.
Do you thing that issue related DNS? 
> The EAI_FAIL is related to not being able to resolve the DNS. https://stackoverflow.com/a/70714669

```
{"message":"getaddrinfo EAI_FAIL admin:admin@localhost","stack":"Error: getaddrinfo EAI_FAIL admin:admin@localhost\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26)","errno":-3004,"code":"EAI_FAIL","syscall":"getaddrinfo","hostname":"admin:admin@localhost"}
```

**Comment by pieper** [comment]
That sounds likely, yes.  It looks like you are on windows, and I'm not sure that this server has ever been tested there.  You could try creating a small linux VM to test how things are meant to work and then see which steps are different on windows.

**Comment by taivnn9** [comment]
Yes, thank . I tried on MacOS have same error. do you have work on this code. 
could you provide to me version of [Apache CouchDB](http://couchdb.apache.org/) you're using it.

Maybe version related? Both my Windows & Mac using v. 3.3.1.
<img width="899" alt="Screen Shot 2023-03-19 at 16 30 56" src="https://user-images.githubusercontent.com/24297201/226166131-f0914f0f-eeeb-4b11-aef6-effa823b8eca.png">

**Comment by pieper** [comment]
To be honest, I haven't run the server myself in a while and I guess we didn't track the couchdb version, but it would have been whatever was current at the time.  It looks like you are on the right track to solving this and if it's a version issue or something else we need to document a PR would be very welcome.

**Comment by taivnn9** [comment]
Yes. Thank a lot for your help. Let me find out root cause

Vào 01:38, T.2, 20 Th3, 2023 Steve Pieper ***@***.***> đã
viết:

**Comment by emelalkim** [comment]
hi @taivnn9, sorry for the late response.
We are using the dockerized version of the server with ePad. I'd be happy to try one of your DICOMs if you can share a deidentified version with me.
Currently, we use ibmcom/couchdb3:latest version of dockerized couchdb, but other version 3 couchdb should be fine. 
Are you using dicomweb-server in the dockerized form? If yes, you shouldn't use localhost. inside the dicomweb-server container, localhost resolves to the container, not to your host.
